crates/wasmtime/src/module.rs

```diff
@@ -309,6 +309,30 @@ impl Module {
         })
     }
 
+    /// Read compiled module from I/O.
+    pub unsafe fn deserialize(engine: &Engine, r: impl std::io::Read) -> Result<Module> {
```

Comment:
Is this intended to be `unsafe`? Ideally we'd have this be a safe function I think.

Reply:
Also like serialize I think it may be best to be conservative and start out with a byte slice instead of a `Read`

Reply:
> Is this intended to be `unsafe`? Ideally we'd have this be a safe function I think.

I mentioned that it is marked unsafe since it does not provide any guarantees for safety of the ingested code. We don't want deal with code signing and currently rely on caller to that.

Reply:
I wonder if we really want to mark this as unsafe. I get the intent behind it, but it's worth noting that the current API that transparently loads a cache entry if that exists has the same issue: since anybody could put whatever they want into the cache, the same "unsafety" exists.

@sunfishcode what's your take on this, do we have enough of a secure setup in place for the case where we are fully in control of the pipeline from compiling a wasm file to running it that we should call out the gap in safety like this?

Reply:
Oh sorry I missed the comment in the PR description! I would personally feel though that we shouldn't have `unsafe` here. I think it's a good point to document that arbitrary bytes here is not truly safe and this shouldn't be fed untrusted code, but that feels like such a far away enough use case that it's not worth making this `unsafe` as a result.